src/main/java/ch/njol/skript/expressions/ExprExcept.java

```diff
@@ -0,0 +1,87 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.comparator.Comparators;
+import org.skriptlang.skript.lang.comparator.Relation;
+
+@Name("Except")
+@Description("Filter a list by providing objects to be excluded.")
+@Example("""
+	spawn zombie at location(0, 0, 0):
+		hide entity from all players except {_player}
+	""")
+@Example("""
+	set {_items::*} to a copper ingot, an iron ingot and a gold ingot
+	set {_except::*} to {_items::*} excluding copper ingot
+	""")
+@Since("INSERT VERSION")
+public class ExprExcept extends SimpleExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprExcept.class, Object.class, ExpressionType.COMBINED,
+				"%~objects% (except|excluding|not including) %objects%");
+	}
+
+	private Expression<?> source;
+	private Expression<?> exclude;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		source = LiteralUtils.defendExpression(exprs[0]);
+		if (source.isSingle()) {
+			Skript.error("Must provide a list containing more than one object to exclude objects from.");
+			return false;
+		}
+		exclude = LiteralUtils.defendExpression(exprs[1]);
+		return LiteralUtils.canInitSafely(source, exclude);
+	}
+
+	@Override
+	protected Object @Nullable [] get(Event event) {
+		Object[] exclude = this.exclude.getArray(event);
+		if (exclude == null || exclude.length == 0)
+			return source.getArray(event);
+
+		return source.streamAll(event)
```

Comment:
This should probably be an expression that primarily implements iterator() instead of get(), since i imagine a large amount of its use will be with loops. See ExprTimes for an example.

That may conflict with the whole getAll thing, so if it does, just leave it as is for now.